src/main/java/teammates/common/util/Version.java

```diff
@@ -0,0 +1,152 @@
+package teammates.common.util;
+
+/**
+ * Represents a version by 3 parts: major version, minor version and patch version.
+ */
+public class Version implements Comparable<Version> {
+    /**
+     * The original String of the version. It could be either XX-XX-XXXXX or XX.XX.XXXX format.
+     */
+    private String originalRepresentation;
+    private Integer major;
+    private Integer minor;
+    private String patch;
```

Comment:
then the version 4.44 will come before 10.44 but it should be 10.44 that goes first. The major part at least should be Integer because the length of the number may not be the same.